src/lib/edge-functions/registry.mjs

```diff
@@ -321,50 +377,69 @@ export class EdgeFunctionsRegistry {
   }
 
   /**
-   * @param {EdgeFunction} func
+   * Logs an event associated with functions.
+   *
+   * @param {EdgeFunctionEvent} event
+   * @param {object} data
+   * @param {Error} [data.buildError]
+   * @param {string} [data.functionName]
+   * @param {string[]} [data.warnings]
+   * @returns
    */
-  #logAddedFunction(func) {
-    log(`${NETLIFYDEVLOG} ${chalk.green('Loaded')} edge function ${chalk.yellow(this.#getDisplayName(func.name))}`)
-  }
+  #logEvent(event, { buildError, functionName, warnings = [] }) {
+    const subject = functionName
+      ? `edge function ${chalk.yellow(this.#getDisplayName(functionName))}`
+      : 'edge functions'
+    const warningsText =
+      warnings.length === 0 ? '' : ` with warnings:\n${warnings.map((warning) => `  - ${warning}`).join('\n')}`
 
-  /**
-   * @param {EdgeFunction} func
-   */
-  #logDeletedFunction(func) {
-    log(`${NETLIFYDEVLOG} ${chalk.magenta('Removed')} edge function ${chalk.yellow(this.#getDisplayName(func.name))}`)
+    if (event === 'buildError') {
+      log(`${NETLIFYDEVERR} ${chalk.red('Failed to load')} ${subject}: ${buildError}`)
+
+      return
+    }
+
+    if (event === 'loaded') {
+      const icon = warningsText ? NETLIFYDEVWARN : NETLIFYDEVLOG
+      const color = warningsText ? chalk.yellow : chalk.green
+
+      log(`${icon} ${color('Loaded')} ${subject}${warningsText}`)
+
+      return
+    }
+
+    if (event === 'reloaded') {
+      const icon = warningsText ? NETLIFYDEVWARN : NETLIFYDEVLOG
+      const color = warningsText ? chalk.yellow : chalk.green
+
+      log(`${icon} ${color('Reloaded')} ${subject}${warningsText}`)
+
+      return
+    }
+
+    if (event === 'reloading') {
+      log(`${NETLIFYDEVLOG} ${chalk.magenta('Reloading')} ${subject}...`)
+
+      return
+    }
+
+    if (event === 'removed') {
+      log(`${NETLIFYDEVLOG} ${chalk.magenta('Removed')} ${subject}`)
+    }
   }
 
   /**
    * @param {string} urlPath
    * @param {string} method
    */
   matchURLPath(urlPath, method) {
-    const declarations = this.#bundler.mergeDeclarations(
```

Comment:
Instead of generating a manifest and routes on each request, we generate it every time we build the functions and then cache it in `this.#manifest` and `this.#routes`.